### PR TITLE
Remove pcre dependency by using regex instead of nre

### PR DESCRIPTION
**Pull request by hendrikgit**
Addresses issue https://github.com/docopt/docopt.nim/issues/33.

This replaces the [nre](https://nim-lang.org/docs/nre.html) module with [regex](https://github.com/nitely/nim-regex). nitely recently added [support for split with captures](https://github.com/nitely/nim-regex/issues/8) with the `splitIncl` proc.

It looks like thanks to nitelys work only these small changes are needed now to replace nre, the tests all passed.

### CODE REVIEW

File: src/docopt.nim [hunk 1]
```diff
@@ -532,7 +533,7 @@ proc parse_argv(tokens: TokenStream, options: var seq[Option],
 
 
 proc parse_defaults(doc: string): seq[Option] =
-    var split = doc.split(re"\n\ *(<\S+?>|-\S+?)")
+    var split = doc.splitIncl(re"\n\ *(<\S+?>|-\S+?)")
```
Comment (oprypin) [inline src/docopt.nim]: `mixedCase` not allowed

**Comment by data-man** [comment]
Why not stdlib's pegs?

File: docopt.nimble [hunk 2]
```diff
@@ -5,6 +5,7 @@ license = "MIT"
 srcDir = "src"
 
 requires "nim >= 0.15.0"
+requires "regex >= 0.7.4"
 
 task test, "Test":
   exec "nim compile --verbosity:0 --run test/test"
```
Comment (dom96) [inline docopt.nimble]: Change this to ``exec "nimble c --verbosity:0 -r -y test/test`` and you'll be able to get rid of the ``nimble`` lines you've added in the travis file. Repeating what is inside this .nimble file should never be necessary.

**Comment by hendrikgit** [comment]
Thank you @dom96. Do you have an idea about the `Error: Nimble cannot find Nim's standard library.` issue? I tried setting the NIM_LIB_PREFIX but it did not help.

**Comment by dom96** [comment]
> You can set the environment variable NIM_LIB_PREFIX to where Nim's `lib` directory is located as a workaround.

Set it to where the *`lib` directory is*, not to the `lib` directory. So: ``nim-master/`` I guess, it might be necessary to use ``$PWD/nim-master``

**Comment by LemonBoy** [comment]
Friendly ping.

**Comment by brentp** [comment]
I have this passing in #44. I have just pushed the request to change `splitIncl` to `split_incl`.
Is there anything else needed for this?

If not, I can make make the PR to the branch here or we can use #44.

**Comment by oprypin** [comment]
Great to hear that it passes!

Though I'm a bit stumped by this dilemma: I'd have to either submit broken (but then immediately fixed) commits into the repo or mess with the authorship info somehow.

**Comment by oprypin** [comment]
Thanks!

**Pull request merged**